broadinstitute/single_cell_portal
# Publish single_cell_portal to PyPI (SCP-2414)

Update manage_study.py to run as an executable (manage-study) after installation of the single_cell_portal package from PyPI.
Minor updates to repo (setup.py, __init__.py etc) for publishing to PyPI.

Currently single_cell_portal is published to TestPyPI, to install:
`pip install --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple  single_cell_portal`

### Expected current behavior:

single_cell_portal installs scp-ingest-pipeline - the available version is 1.3.7 which will fail on .txt files with "_csv.Error: Could not determine delimiter"

Running manage-study against the staging server with a valid .txt metadata file against the metadata convention will fail (due to local 1.3.7 code) even though the staging server is running ingest 1.3.9

### To test:
(First, rename a synthetic metadata file from .tsv to .txt)

```
gcloud config configurations activate default

ACCESS_TOKEN=`gcloud auth print-access-token`

STUDY_NAME=<name of a staging study w/o a current metadata file>

manage-study --environment staging --token=$ACCESS_TOKEN  upload-metadata --file metadata.txt --study-name $STUDY_NAME --use-convention
```
running with the same file saved as metadata.tsv should succeed

These changes support SCP-2414.
